addons/web/controllers/main.py

```diff
@@ -860,10 +852,11 @@ def _login_redirect(self, uid, redirect=None):
 
     @http.route('/web/login', type='http', auth="none")
     def web_login(self, redirect=None, **kw):
+        allow_open_redirect = not request.env['ir.config_parameter'].sudo().get_param('web.login.noredirect', False)
```

Comment:
Let's drop this and use the default `local=True` here as for all `request.redirect()` calls. We'll see where we need to allow external redirect explicitly at login (probably only for our OAuth2 provider implementation)